roles/openshift_health_checker/openshift_checks/docker_storage.py

```diff
@@ -0,0 +1,181 @@
+                    "Use of loopback devices with the Docker devicemapper storage driver\n"
+                    "(the default storage configuration) is unsupported in production.\n"
+                    "Please use docker-storage-setup to configure a backing storage volume.\n"
+                    "See http://red.ht/2rNperO for further information."
+                )
+                return {"failed": True, "changed": changed, "msg": msg}
+            result = self._check_dm_usage(driver_status, task_vars)
+            result["changed"] = changed
+            return result
+
+        # TODO(lmeyer): determine how to check usage for overlay2
+
+        return {"changed": changed}
+
+    @staticmethod
+    def _is_dm_using_loopback_device(driver_status):
+        return bool(driver_status.get("Data loop file"))
+
+    def _check_dm_usage(self, driver_status, task_vars):
+        """
+        For background: under thinpool-backed devicemapper, docker-storage-setup creates
+        an LVM2 LV as a backing storage pool. The LV is "thin" because it does not use
+        all available storage from its VG, instead expanding as needed; so in considering
+        available space, it is necessary to include space remaining in the VG (which is
+        unfortunately not reported by the storage driver in docker info).
+        Usage of the LV is divided into space allocated to data and metadata (which also
+        expand automatically), either of which could run out of space while the other is
+        fine; so both must be checked.
+        """
+        vals = dict(
+            vg_free=self._get_vg_free(driver_status.get("Pool Name"), task_vars),
+            data_used=driver_status.get("Data Space Used"),
+            data_total=driver_status.get("Data Space Total"),
+            metadata_used=driver_status.get("Metadata Space Used"),
+            metadata_total=driver_status.get("Metadata Space Total"),
+        )
+
+        for key, value in vals.copy().items():
+            try:
+                vals[key + "_bytes"] = self._convert_to_bytes(value)
+            except ValueError as err:  # unlikely to see this from our inputs
+                return {
+                    "failed": True,
+                    "values": vals,
+                    "msg": "Could not interpret {} value '{}' as bytes: {}".format(key, value, str(err))
+                }
+
+        try:
+            vals["data_threshold"] = float(get_var(task_vars, "max_thinpool_data_usage_percent",
+                                                   default=self.max_thinpool_data_usage_percent))
+            vals["metadata_threshold"] = float(get_var(task_vars, "max_thinpool_metadata_usage_percent",
+                                                       default=self.max_thinpool_meta_usage_percent))
+        except ValueError as err:
+            return {
+                "failed": True,
+                "msg": "Unable to convert thinpool data usage limit to float: {}".format(str(err))
```

Comment:
the bad value is included in the error which is included in the msg. you're right it doesn't tell you which value failed, but i thought the user could figure it out from that. but i suppose that's just lazy; I just didn't want to duplicate the entire try/except for two almost identical things